pkg/controller/clusterinst_api.go

```diff
@@ -892,61 +968,78 @@ func (s *ClusterInstApi) createClusterInstInternal(cctx *CallContext, in *edgepr
+			if err != nil {
+				return fmt.Errorf("failed to select infra flavor, %s", err)
+			}
+			in.OptRes = optRes
+			in.AvailabilityZone = az
+			log.SpanLog(ctx, log.DebugLevelApi, "Selected Cloudlet Node Flavor", "res", in.NodeResources)
+		}
+		// Handle control (master) node specification
+		// Note for platforms that manage the control nodes themselves
+		// like Azure AKS, etc, we don't need to configure any master
+		// node resources.
+		if in.Deployment == cloudcommon.DeploymentTypeKubernetes && !features.ManagesK8SControlNodes && !features.NoClusterSupport {
+			// For platforms that do not manage the master nodes,
+			// we need to set the master node flavor so we can create
+			// the master nodes.
+			// Note: temporarily drop support for running workloads on
+			// master nodes. To do so we should allow one of the user
+			// specified node pools to become the master pool.
+			// That requires more work in the platform code to support
+			// such a change and allow for more than 1 master node.
+			var masterNodeResources *edgeproto.NodeResources
+			settings := s.all.settingsApi.Get()
+			if settings.MasterNodeFlavor != "" {
 				masterFlavor := edgeproto.Flavor{}
 				masterFlavorKey := edgeproto.FlavorKey{}
-				settings := s.all.settingsApi.Get()
 				masterFlavorKey.Name = settings.MasterNodeFlavor
-
 				if s.all.flavorApi.store.STMGet(stm, &masterFlavorKey, &masterFlavor) {
-					log.SpanLog(ctx, log.DebugLevelApi, "MasterNodeFlavor found ", "MasterNodeFlavor", settings.MasterNodeFlavor)
-					vmspec, err := s.all.resTagTableApi.GetVMSpec(ctx, stm, masterFlavor, in.MasterNodeFlavor, cloudlet, info)
-					if err != nil {
-						// Unlikely with reasonably modest settings.MasterNodeFlavor sized flavor
-						log.SpanLog(ctx, log.DebugLevelApi, "Error K8s Master Node Flavor matches no eixsting OS flavor", "nodeFlavor", in.NodeFlavor)
-						return err
-					} else {
-						in.MasterNodeFlavor = vmspec.FlavorName
-						log.SpanLog(ctx, log.DebugLevelApi, "Selected Cloudlet Master Node Flavor", "vmspec", vmspec, "master flavor", in.MasterNodeFlavor)
-					}
+					masterNodeResources = masterFlavor.ToNodeResources()
 				} else {
-					// should never be non empty and not found due to validation in update
-					// revert to using NodeFlavor (pre EC-1767) and log warning
-					in.MasterNodeFlavor = in.NodeFlavor
-					log.SpanLog(ctx, log.DebugLevelApi, "Warning : Master Node Flavor does not exist using", "master flavor", in.MasterNodeFlavor)
+					return fmt.Errorf("default master node flavor %q in settings is not found, please contact your administrator", settings.MasterNodeFlavor)
 				}
+			} else {
+				log.SpanLog(ctx, log.DebugLevelApi, "using default master node resources")
+				// no master node flavor specified by admin, just
+				// use minimum resources.
+				masterNodeResources = &edgeproto.NodeResources{
+					Vcpus: 1,
+					Ram:   2048,
+					Disk:  10,
+				}
```

Comment:
Do you think this is sufficient to run control plane? For the disk I think we need to have 20G as I don't remember if I optimized the latest base image to fit into 10G

Reply:
Filed this issue to track: https://github.com/edgexr/edge-cloud-platform/issues/379